playbooks/operationalizing/monitoring_guide.adoc

```diff
@@ -0,0 +1,99 @@
+
+|Docker Metadata Storage |Check that docker's metadata storage volume is not full | `echo "$(lvs \| awk '/docker-pool/ {print $6}') > 70" \| bc`
+
+|=========================================================
+
+=== Nodes & Masters
+
+.Node,Master checks
+[width="100%",cols="3,6,2,10",options="header"]
+|=========================================================
+|Check Name |Description | Relevant Hosts| Sample Alerting Logic
+
+|Etcd Service |Check that etcd is active |Masters | `systemctl is-active etcd`
+
+|Etcd Storage |Check that the etcd volume is not too full |Masters | `echo "$(lvs \| awk '/etcd/ {print $5}') > 70" \| bc`
+
+|Master API Service (single master) |Check that the Master API Service is active|Masters | `systemctl is-active atomic-openshift-master`
+
+|Master API Service (multi-master) |Check that the Master API Service is active|Masters | `systemctl is-active atomic-openshift-master-api`
+
+|Master Controllers Service (multi-master) | Check that the Master Controllers Service is active|Masters | `systemctl is-active atomic-openshift-master-controllers`
+
+|Node Service |Check that the node service is active | All Nodes| `systemctl is-active atomic-openshift-node`
+
+|Node Storage |Check that the node's local data storage volume is not too full | All Nodes | `echo "$(lvs \| awk '/origin/ {print $5}') > 70" \| bc`
+
+|OpenVSwitch Service |Check that the openvswitch service is active | All Nodes| `systemctl is-active openvswitch`
+|=========================================================
+
+=== API Endpoints
+
+Many OpenShift components expose HTTP based endpoints for interrogating the health and current operation. The following endpoints should be monitored.
+
+.API Endpoint checks
+[width="100%",cols="3,6,10",options="header"]
+|=========================================================
+|Check Name |Description |Sample Alerting Logic
+
+|OpenShift Master API Server |Check the health of a master API Endpoint| `curl -H "Authorization: Bearer $(oc whoami -t)" https://console.c1-ocp.myorg.com:8443/healthz \| grep ok`
+
+|Router |Check the health of the Router| `curl http://router.default.svc.cluster.local:1936/healthz \| grep 200`
+
+|Registry |Check the health of the Registry| `curl -I https://docker-registry.default.svc.cluster.local:5000/healthz \| grep 200`
+
+|Logging | Check the health of the EFK Logging Stack | https://github.com/redhat-cop/openshift-toolkit/blob/master/health_check/elasticsearch-health-check-ocp34.sh
+
+|Metrics | Check the health of the Metrics Stack | https://github.com/redhat-cop/openshift-toolkit/blob/master/health_check/metrics-health-check.sh
+
+|=========================================================
+
+== Ensuring a cluster has adequate capacity
+
+.Feedback or Contribution Needed
+****
+Need a cluster capacity monitoring guide.
+****
+
+== What's Next?
+
+* Monitoring Tooling Implementations (Coming Soon!)
```

Comment:
A partial implementation for Nagios can be found here:
https://github.com/redhat-cop/casl-ansible/tree/master/roles/nagios-target
(includes more than just OpenShift)